src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -85,18 +85,39 @@ internal static class ThrowHelper {
                                                     Environment.GetResourceString(GetResourceName(resource)));
         }
 
+        internal static void ThrowArgumentOutOfRangeException(ExceptionArgument argument, int paramNumber, ExceptionResource resource)
+        {
+            throw new ArgumentOutOfRangeException(GetArgumentName(argument) + "[" + paramNumber + ']',
```

Comment:
I realize this is an error path, so extra allocations aren't as much of a concern, but the `paramName` could be passed to `ArgumentOutOfRangeException` as:

```c#
GetArgumentName(argument) + "[" + paramNumber.ToString() + "]"
```

 - `paramNumber` -> `paramNumber.ToString()` avoids a box allocation and avoids the `object[]` array allocation with the call to `string.Concat(object[])` (`string.Concat(string,string,string,string)` will be used instead).
 - `']'` -> `"]"` isn't strictly necessary as Roslyn is smart enough to treat this as a string constant, instead of as a char that will box and cause `string.Concat(object[])` to be called, but is for consistency with earlier `"["`.